### ISSUE TITLE
first-timers-only: Add test to make sure pausing while speech recognition is off works properly

### ISSUE THREAD

**Issue by TalAter** [TalAter/annyang#165]
#### first-timers-only

This issue is tagged :octocat: **first-timers-only**. It is only for people who have never contributed to open source before, and are looking for an easy way take their first steps.

Thank you for your help :heart:

#### Start / Abort vs. Pause / Resume

There are four ways to change annyang's listening status:

- `annyang.start` - Starts the browser's SpeechRecognition engine, and makes annyang respond to voice commands.
- `annyang.abort` - Stops the browser's SpeechRecognition engine, and makes annyang stop responding to voice commands.
- `annyang.pause` - Does not stop the browser's Speech Recognition engine if it is on, but instructs annyang not to respond to voice commands.
- `annyang.resume` - Starts the browser's SpeechRecognition engine if it was aborted, and makes annyang respond to voice commands if it was paused.

This allows a developer to either shut off Speech Recognition and annyang completely with `abort`, or just `pause` annyang temporarily, so that when annyang is `resume`d, restarting the browser's Speech Recognition isn't required.

#### The Issue

The test specs do not currently test to verify that if `annyang.pause()` is called after an `annyang.abort()` then:

- Speech recognition remains off.
- annyang remains paused.

Add tests that verify these specs:
````javascript
  it('should leave annyang paused if called after annyang.abort()', func1);
  it('should leave the browser\' Speech Recognition off, if called after annyang.abort()', func1);
}
````

The tests described inside `describe('annyang.pause', function() {…});` in *test/spec/BasicSpec.js*, should be used as a template for adding the two new specs described above.

Speech Recognition can be tested with: `recognition.isStarted()`.
annyang can be tested with: `annyang.isListening()`

#### Contributing to annyang

- [ ] Fork the repository from the [annyang GitHub page](https://github.com/TalAter/annyang).
- [ ] Clone a copy to your local machine with `$ git clone git@github.com:YOUR-GITHUB-USER-NAME/annyang.git`
- [ ] Make sure you have *node.js* and *npm* installed on your machine. You can use this [guide](https://docs.npmjs.com/getting-started/installing-node) for help.
- [ ] Install all of annyang's development dependencies with npm. `$ cd annyang; npm install`
- [ ] Run grunt to make sure everything runs smoothly `$ grunt`
- [ ] Code, code, code.
- [ ] Run `$ grunt` after making changes to verify that everything still works and run the build process. Alternatively, you can just run `$ grunt watch` once, and leave that process running. It will continuously run all the tests and build the files every time you make a change to one of annyang's files. It will even *beep* if you make an error, and help you debug it.:+1:
- [ ] Before committing your changes, the last step must always be running `$ grunt`. This makes sure everything works, and all files are kept up to date with your changes.
- [ ] Once you've made sure all your changes work correctly, and committed all your changes, push your local changes back to github with `$ git push -u origin master`
- [ ] Visit your fork on GitHub.com ([https://github.com/YOUR-USER-NAME/annyang](https://github.com/YOUR-USER-NAME/annyang)) and create a pull request for your changes.
- [ ] Makes sure your pull request describes exactly what you changed and references this issue (just include the issue number in the title like this: #49)

#### About the tests

annyang is tested using [Jasmine](http://jasmine.github.io/2.0/introduction.html).

BasicSpec.js contains a series of spec groups (e.g. `describe('a spec group', function() {})`) which each contain 1 or more specs (e.g. `it('should do stuff', function() {})`). Some of the spec groups also contain some code which runs before each spec (`beforeEach(function() {})`).

To simulate Speech Recognition in the testing environment, annyang uses a mock object called [Corti](https://github.com/TalAter/Corti) which mocks the browser's SpeechRecognition object. Corti also adds a number of utility functions to the SpeechRecognition object which simulate user actions (e.g. `say('Hello there')`), and allow checking the SpeechRecognition's status (e.g. `isListening() === true`).

**Comment by Norybdigital** [comment]
Hello!  
I just submitted my first github pull request.  I hope I followed procedure correctly.  I really like this project and look forward to working on/with it in the future!

**Comment by TalAter** [comment]
Fixed by @Norybdigital in pull request #166 

Congrats on your first :octocat: pull request, and thank you for the help :heart:
